python_modules/libraries/dagster-components/dagster_components/core/component.py

```diff
@@ -43,6 +45,9 @@ def get_source_position_tree(self) -> Optional[SourcePositionTree]: ...
 
 @scaffold_with(DefaultComponentScaffolder)
 class Component(ABC):
+    @classmethod
+    def __dg_library_object__(cls) -> None: ...
```

Comment:
Maybe changed upstack but seems odd for this to be a class method instead of an attr

Reply:
yeah the reason I did this was that our `Resolvable` machinations treat these dunder fields as regular fields that are part of the schema if you just make it a regular classvar

so just wanted to avoid that issue for this PR